api/v1beta1/integrationtestscenario_types.go

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	applicationapiv1alpha1 "github.com/redhat-appstudio/application-api/api/v1alpha1"
+	tektonv1beta1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// IntegrationTestScenarioSpec defines the desired state of IntegrationScenario
+type IntegrationTestScenarioSpec struct {
+	// Application that's associated with the IntegrationTestScenario
+	// +kubebuilder:validation:Pattern=^[a-z0-9]([-a-z0-9]*[a-z0-9])?$
+	// +required
+	Application string `json:"application"`
+	// Tekton Resolver where to store the Tekton resolverRef trigger Tekton pipeline used to refer to a Pipeline or Task in a remote location like a git repo.
+	// +required
+	ResolverRef tektonv1beta1.ResolverRef `json:"resolverRef"`
```

Comment:
General word of advice when it comes to APIs - don't reference other project CRDs (objects and sub-objects). APIs can change in unexpected ways, especially if they are not labeled `v1`. You are better off defining your own object here, using only the fields you care about. This advice also applies to k8s objects.

Reply:
Thanks. Let's use our own object. That works, but it is strange why referencing tekton.ResolverRef will fails in unittest.